pkg/mcshub/controller/controller_test.go

```diff
@@ -97,411 +82,291 @@ func TestImportGenerated(t *testing.T) {
-// I think it should delete the service import too since its owner is the serviceExport that was deleted.
-func TestCreateAndDeleteExport(t *testing.T) {
-	assertions := require.New(t)
-	exp, es := prepareServiceExport(t)
-	preloadedObjects := []runtime.Object{service}
+// tests that change in non essential export properties don't change the import
+func TestUpdateExportWithoutImport(t *testing.T) {
+	assert := require.New(t)
+	exp, err := prepareServiceExport(export, cluster1)
+	assert.Nil(err)
+
+	preloadedObjects := []runtime.Object{exp}
 	ser := controller.ServiceExportReconciler{
 		Client: getClient(preloadedObjects),
 		Log:    newLogger(t, false),
 		Scheme: getScheme(),
 	}
-	ser.Client.Create(context.TODO(), exp)
-	req1 := reconcile.Request{
+	req := reconcile.Request{
 		NamespacedName: types.NamespacedName{
 			Name:      exp.GetName(),
 			Namespace: exp.GetNamespace(),
 		}}
-	result, err := ser.Reconcile(context.TODO(), req1)
 
-	if err != nil {
-		t.Error(err)
-	}
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
+	_, err = ser.Reconcile(context.TODO(), req)
+	assert.Nil(err)
 
-	si := mcsv1a1.ServiceImport{}
-	err = ser.Client.Get(context.TODO(), req1.NamespacedName, &si)
+	// save the generated import
+	pre := &mcsv1a1.ServiceImport{}
+	err = ser.Client.Get(context.TODO(), req.NamespacedName, pre)
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
-	}
-	// TO PR - I wanted to test "as close" to reality, and get it from the es and not initialized those values
-	// maybe its not needed, and can be initialized
-	excepted := generateExceptedServiceImport(exp, es)
-	compareSi(&excepted, &si, assertions)
+	// modify the export's label and reconcile again
+	exp = &mcsv1a1.ServiceExport{}
+	err = ser.Client.Get(context.TODO(), req.NamespacedName, exp)
+	assert.Nil(err)
+	exp.GetLabels()["app"] = "foo"
+	err = ser.Client.Update(context.TODO(), exp, &client.UpdateOptions{})
+	assert.Nil(err)
 
-	//delete the export - and check that the import was also deleted:
-	ser.Client.Delete(context.TODO(), exp)
-	result, err = ser.Reconcile(context.TODO(), req1)
+	_, err = ser.Reconcile(context.TODO(), req)
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
-	}
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
-	si = mcsv1a1.ServiceImport{}
-	err = ser.Client.Get(context.TODO(), req1.NamespacedName, &si)
-	assertions.True(apierrors.IsNotFound(err))
+	post := &mcsv1a1.ServiceImport{}
+	err = ser.Client.Get(context.TODO(), req.NamespacedName, post)
+	assert.Nil(err)
+
+	assert.Equal(pre.Spec, post.Spec)
+	// @todo: can't compare entire object as resource version changes (reconcile always calls update)
 }
 
-// not finished - need to change some field (!)
-func TestUpdateExportWithoutImport(t *testing.T) {
-	exp1, _ := prepareServiceExport(t)
-	preloadedObjects := []runtime.Object{service, exp1}
+// tests that change in non essential export properties don't change the import
```

Comment:
comment here is not correctly describing the test

Reply:
cut and paste from previous. Fixed